src/main.rs

```diff
@@ -0,0 +1,482 @@
+                        Sample rate: {:.0}\n\
+                        Target frequency: {}\n\
+                        Detected frequency: {:.1}hz\n\
+                        ",
+                        device.name().unwrap(),
+                        state.sample_rate,
+                        match state.target_frequency {
+                            Some(f) => format!("{:.1}hz", f),
+                            None => "Not selected".into(),
+                        },
+                        avg_f0,
+                    )),
+                    info_layout[0],
+                );
+
+                let is_inside = state.cursor_pos.0 >= main_layout[2].top()
+                    && state.cursor_pos.0 < main_layout[2].bottom()
+                    && state.cursor_pos.1 >= main_layout[2].left()
+                    && state.cursor_pos.1 < main_layout[2].right();
+
+                if is_inside && state.mouse_pressed {
+                    let alpha = (state.cursor_pos.1 - main_layout[2].left()) as f32
+                        / (main_layout[2].width - 1) as f32;
+                    let (low, high) = match state.graph {
+                        Graph::AmplitudeSpectrum => (
+                            f32::ln(state.sample_rate / FFT_SIZE as f32),
+                            f32::ln(state.sample_rate / 2.0),
+                        ),
+                        Graph::CMNDF => (
+                            f32::ln(state.sample_rate / (cmndf.len() - 1) as f32),
+                            f32::ln(state.sample_rate / 2.0),
+                        ),
+                    };
+                    let log_x = lerp(low, high, alpha);
+                    state.target_frequency = Some(f32::exp(log_x));
+                }
+            })?;
+
+            loop {
+                match event::poll(Duration::from_millis(0)) {
+                    Ok(ready) if ready => {
+                        let ev = event::read().unwrap();
+                        match ev {
+                            Event::Key(key) => match key.code {
+                                event::KeyCode::Char('c' | 'C')
+                                    if key.modifiers == KeyModifiers::CONTROL =>
+                                {
+                                    state.running = false;
+                                }
+                                event::KeyCode::Char('w') if key.kind == KeyEventKind::Press => {
+                                    state.apply_window = !state.apply_window;
+                                }
+                                event::KeyCode::Char(note_char)
+                                    if ('a'..='g').contains(&note_char)
+                                        && key.kind == KeyEventKind::Press =>
+                                {
+                                    let semitones = [0, 2, -9, -7, -5, -4, -2]
+                                        [note_char as usize - 'a' as usize];
```

Comment:
```suggestion
                                    let semitones = [0, 2, -9, -7, -5, -4, -2]
                                        [('a'..='g').position(|char| char == note_char).unwrap()];
```
I'm not sure, but for me it looks nicer than substracting chars as usize

If I understand correctly it can be replaced. I guess it is easier to understand